lifetimes/estimation.py

```diff
@@ -71,15 +69,16 @@ def _loglikelihood(params, x, tx, T):
 
         recency_T = T - tx - 1
 
+        js = np.arange(max(recency_T) + 1)
+        @np.vectorize
         def _sum(x, tx, recency_T):
-
-            j = np.arange(recency_T+1)
+            j = js[:recency_T + 1]
```

Comment:
This was doing n^2 memory allocations. Now the values are stored in memory once which should cut the cost of computing loglikelihood by it's square root.